uberfire-io/src/main/java/org/uberfire/io/lock/BatchLockControl.java

```diff
@@ -27,24 +29,25 @@
     private FileSystem fileSystemOnBatch;
 
     public void lock(final FileSystem fs) {
+        checkNotNull("fs", fs);
         lock.lock();
         try{
             makeSureThatIsOnlyOneFSOnCurrentBatch(fs);
+
+            if (!isAlreadyOnBatch(fs)) {
```

Comment:
I moved this block here so that in the event of an unlockable filesystem being passed into the method, `lock.unlock()` would still be called.

Would appreciate a sanity check, @ederign

Reply:
Looks good!